parsl/dataflow/dflow.py

```diff
@@ -167,13 +173,6 @@ def __init__(self, config=Config()):
                 logger.error("invalid checkpoint_period provided: {0} expected HH:MM:SS".format(config.checkpoint_period))
                 self._checkpoint_timer = Timer(self.checkpoint, interval=(30 * 60), name="Checkpoint")
 
-        # if we use the functionality of dynamically adding executors
-        # all executors should be managed.
-        if any([x.managed for x in config.executors]):
-            self.flowcontrol = FlowControl(self)
-        else:
-            self.flowcontrol = FlowNoControl(self)
-
```

Comment:
I think it's fine to get rid of FlowNoControl entirely like this -- the flow control/strategy stuff has to cope with a mix of managed and unmanaged executors in the FlowControl case anyway. It leads to the flow control stuff running even when running with just a local thread executor, but I don't think that is a huge price to pay for this simplification.

Getting rid of FlowNoControl could be merged independently of the rest of this PR.

Reply:
Code can be added to disable it if there is nothing to poll if executors cannot be added at runtime. But the existence of add_executors() seems to suggest that this is not the intent.